### PR TITLE
Add notifications; render as pop-ups ("toasts")

### LINKED ISSUE #207: Error messages are not user friendly
## Problem

When a user tries to create a post with invalid content, the error message displayed is cryptic and does not contain clear instructions (e.g., `({:path [:post/md-content 1])` instead of `Post must start with a title`). Likewise, when there is a website error (such as a connection timeout), the error is not clearly shown or explained.

## Suggestion

- Add pop-up ("toast") notifications to notify the user of website errors.
    A new dependency (e.g., `react-toastify`) may be needed.

- Add user-friendly error messages to explain what the error is and how to resolve it.

### PR DESCRIPTION
**Pull request by flybot-nam-nguyenhoai**
## Closes #207

This pull request introduces pop-up ("toast") notifications and implements them for website errors (validation and HTTP).

- [x] Add generic notifications (as simple data maps).
    These notify users when their actions succeed or fail, and can be delivered in different ways (e.g., as pop-ups or printed to console output).

- [x] Notify when encountering errors.
    Only validation errors and HTTP failures are notified to users.

- [x] Render notifications as pop-up banners ("toasts").

### Suggestions for future development

- Notify on successful actions.

- Make notification content user friendly.
    For example, error notifications in this PR contain plain Clojure maps of error details—they should instead contain helpful messages in natural language.

### CODE REVIEW

File: client/common/src/flybot/client/common/db/fx.cljs [hunk 1]
```diff
@@ -13,4 +15,28 @@
...
  :fx.log/message
  (fn [messages]
-   (.log js/console (apply str messages))))
\ No newline at end of file
+   (.log js/console (apply str messages))))
+
+;; ---- Toast notifications ----
+
+(rf/reg-fx
+ ;; FIXME: Naming: :fx.ui/ or :fx.app/?
```
Comment (flybot-nam-nguyenhoai) [inline client/common/src/flybot/client/common/db/fx.cljs:23]: Name suggestions are welcome.

File: client/common/src/flybot/client/common/db/event.cljs [hunk 2]
```diff
@@ -302,13 +306,26 @@
  (fn [_ [_ new-options]]
    (edn/read-string new-options)))
 
+;; ------- Notifications ------
+
+(rf/reg-event-fx
+ ;; FIXME: Naming: :evt.ui/ or :evt.app/?
```
Comment (flybot-nam-nguyenhoai) [inline client/common/src/flybot/client/common/db/event.cljs:312]: Name suggestions are welcome.

File: resources/public/index.html [hunk 3]
```diff
@@ -33,6 +33,7 @@
         <meta name="twitter:image" content="http://www.flybot.sg/static/media/flybot-logo.5c92e3d50082cfa03409.png">
 
         <!-- this refers to resources/public/css/style.css -->
+        <link href="/css/ReactToastify.css" rel="stylesheet" type="text/css">
```
Comment (flybot-nam-nguyenhoai) [inline resources/public/index.html:36]: Should this be loaded before or after the main CSS?

File: client/common/src/flybot/client/common/db/event.cljs (hunk 2 above)
Comment (skydread1) [reply client/common/src/flybot/client/common/db/event.cljs:312]: :evt.app/notify maybe?

File: resources/public/index.html (hunk 3 above)
Comment (skydread1) [reply resources/public/index.html:36]: It's fine like that

File: client/web/src/flybot/client/web/core/dom/page.cljs [hunk 4]
```diff
@@ -84,7 +88,9 @@
        (page-post queried-post)
        [:div.post
         [:h2 "No blog posts reside here (yet…)"]
-        [:p "Check your URL while we work on filling up the space here! 🚧 👷 🚧"]])]))
+        [:p "Check your URL while we work on filling up the space here! 🚧 👷 🚧"]])
+     [notifications/toast-notifications]
+     notifications/toast-notification-container]))
```
Comment (skydread1) [inline client/web/src/flybot/client/web/core/dom/page.cljs:93]: No need to call it twice, maybe we can move the component to the header?

**Review by skydread1 — CHANGES_REQUESTED** [review]
Good.

We need to add a cljs test for the notification.

File: client/web/src/flybot/client/web/core/dom/page.cljs (hunk 4 above)
Comment (flybot-nam-nguyenhoai) [reply client/web/src/flybot/client/web/core/dom/page.cljs]: We can make notifications occupy a separate React component, e.g.,
```clojure
;; (ns flybot.client.web.core.dom)

(defn app []
  [:div
   [header-comp]
   [current-page]
   [footer-comp]
   [notification-comp]])
```

Comment (flybot-nam-nguyenhoai) [reply client/web/src/flybot/client/web/core/dom/page.cljs]: Addressed.

File: client/common/src/flybot/client/common/db/fx.cljs (hunk 1 above)
Comment (flybot-nam-nguyenhoai) [reply client/common/src/flybot/client/common/db/fx.cljs]: Decided on `fx.ui`.

File: client/common/src/flybot/client/common/db/event.cljs (hunk 2 above)
Comment (flybot-nam-nguyenhoai) [reply client/common/src/flybot/client/common/db/event.cljs]: Decided on `evt.app`.

**Comment by flybot-nam-nguyenhoai** [comment]
Tests have been added for generic notifications (i.e., notifications that are in plain Clojure map form and stored in `:app/notification`).

**Review by skydread1 — APPROVED** [review]

File: client/common/src/flybot/client/common/db/event.cljs [hunk 5]
```diff
@@ -18,10 +18,14 @@
 
 (rf/reg-event-db
  :fx.http/failure
- [(rf/path :app/errors)]
- (fn [errors [_ result]]
+ (fn [db [_ result]]
     ;; result is a map containing details of the failure
-   (assoc errors :failure-http-result result)))
+   (-> db
+       (assoc-in [:app/errors :failure-http-result] result)
```
Comment (skydread1) [inline client/common/src/flybot/client/common/db/event.cljs:24]: We need to review obsolete error message logic

File: client/web/test/flybot/client/web/core/db_test.cljs [hunk 6]
```diff
@@ -260,4 +260,63 @@
                                 {:posts {:removed-post {:post/id s/post-1-id}}}])))
      (rf/dispatch [:evt.post/remove-post s/post-1-id])
      (testing "Post got removed."
-       (is (= [] @posts))))))
\ No newline at end of file
+       (is (= [] @posts))))))
+
+;; ---------- NOTIFICATIONS ----------
+
+(deftest app-notifications-test
```
Comment (skydread1) [inline client/web/test/flybot/client/web/core/db_test.cljs:267]: Could have been jus a few lines replacing the whole error checks with notification ones

**Review by skydread1 — CHANGES_REQUESTED** [review]

File: client/web/test/flybot/client/web/core/db_test.cljs (hunk 6 above)
Comment (flybot-nam-nguyenhoai) [reply client/web/test/flybot/client/web/core/db_test.cljs]: Simplified. Tests for `:app/errors` and `:app/notification` are now placed next to each other.

File: client/common/src/flybot/client/common/db/event.cljs (hunk 5 above)
Comment (flybot-nam-nguyenhoai) [reply client/common/src/flybot/client/common/db/event.cljs:24]: Will be addressed in a future PR.

**Review by skydread1 — APPROVED** [review]

**Pull request merged**